src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -199,17 +207,15 @@ public Activity AddBaggage(string key, string value)
         public Activity SetParentId(string parentId)
         {
             if (Parent != null)
-                throw new InvalidOperationException(
-                    $"Trying to set {nameof(ParentId)} on activity which has {nameof(Parent)}");
-
-            if (ParentId != null)
-                throw new InvalidOperationException(
-                    $"{nameof(ParentId)} is already set");
-
-            if (string.IsNullOrEmpty(parentId))
-                throw new ArgumentException($"{nameof(parentId)} must not be null or empty");
-
-            ParentId = parentId;
+                NotifyError(new InvalidOperationException($"Trying to set {nameof(ParentId)} on activity which has {nameof(Parent)}"));
```

Comment:
There is certainly a tradeoff between making errors very obvious (fatal), in which case errors are fixed rapidly, and trying to recover from them (in which case they can persist in deployed code).   

This change certainly increases the possibility that programmers will end up with exceptions firing in their code.    This is true however any time you catch an exception.    However there are some things that mitigate that.

1.  System.Diagnostic.Activity is not expected to be used by 'ordinary programmers'.   
2.  When they are used, if these APIs fail you really do have broken logging (probably catastrophically).  Thus you would expect 'normal testing' of their new code to find the issue (and the exceptions just make it easier to find).  
3. Exceptions do show up in both the debugger and in most monitoring systems, so there is visibility into this issue.  It will also show up very nicely in any profiling.    Sure you can miss it, but it is more visible than simply swallowing it.

Finally SOMEONE is going to swallow these exceptions (because we really don't want logging issues to crash our apps).   Thus the problem of having these slip into production remains.  This change at least tries to do it more systematically so that we can be uniform about it.     If we find that this throw-catch ideas is insufficient, we can put the callback event and pipe it to some logging system.